integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/websocket/throttle/WebsocketApiLevelThrottleTestCase.java

```diff
@@ -0,0 +1,132 @@
+                user, storeRestClient);
+        headers.put(TestConstant.AUTHORIZATION_HEADER, "Bearer " + accessToken);
+
+        bandwidthThrottleEndpoint = Utils.getServiceURLWebSocket(BANDWIDTH_THROTTLE_API_CONTEXT + "/" + API_VERSION);
+        eventCountThrottleEndpoint = Utils.getServiceURLWebSocket(EVENT_COUNT_THROTTLE_API_CONTEXT + "/" + API_VERSION);
+    }
+
+    @Test(description = "Test API Level Bandwidth Throttling")
+    public void testAPILevelBandwidthThrottling() throws Exception {
+        HttpResponse api = publisherRestClient.getAPI(bandwidthThrottleApiId);
+        Gson gson = new Gson();
+        APIDTO apidto = gson.fromJson(api.getData(), APIDTO.class);
+        apidto.setApiThrottlingPolicy(BANDWIDTH_THROTTLE_API_POLICY_NAME);
+        APIDTO updatedAPI = publisherRestClient.updateAPI(apidto, bandwidthThrottleApiId);
+        Assert.assertEquals(updatedAPI.getApiThrottlingPolicy(), BANDWIDTH_THROTTLE_API_POLICY_NAME, "API tier not updated.");
+
+        // create Revision and Deploy to Gateway
+        PublisherUtils.createAPIRevisionAndDeploy(bandwidthThrottleApiId, publisherRestClient);
+        Utils.delay(TestConstant.DEPLOYMENT_WAIT_TIME*2, "Couldn't wait until the API was deployed in Choreo Connect");
+
+        WsClient wsClient = new WsClient(bandwidthThrottleEndpoint, headers);
+        String largePayload = "a".repeat(1024);
+        List<String> messages = new ArrayList<>();
+        for (int i = 0; i < 10; i++) {
+            messages.add(largePayload);
+        }
+        messages.add("close");
+        List<String> responses = wsClient.retryConnectUntilDeployed(messages, 0);
+        Assert.assertTrue(responses.size() >= 3 && responses.size() < 10,
+                "Websocket connection was not throttled on bandwidth.");
+    }
+
+    @Test(description = "Test API Level EventCount Throttling")
+    public void testAPILevelEventCountThrottling() throws Exception {
+        HttpResponse api = publisherRestClient.getAPI(eventCountThrottleApiId);
+        Gson gson = new Gson();
+        APIDTO apidto = gson.fromJson(api.getData(), APIDTO.class);
+        apidto.setApiThrottlingPolicy(EVENT_COUNT_THROTTLE_API_POLICY_NAME);
+        APIDTO updatedAPI = publisherRestClient.updateAPI(apidto, eventCountThrottleApiId);
+        Assert.assertEquals(updatedAPI.getApiThrottlingPolicy(), EVENT_COUNT_THROTTLE_API_POLICY_NAME, "API tier not updated.");
+
+        // create Revision and Deploy to Gateway
+        PublisherUtils.createAPIRevisionAndDeploy(eventCountThrottleApiId, publisherRestClient);
+        Utils.delay(TestConstant.DEPLOYMENT_WAIT_TIME*2, "Couldn't wait until the API was deployed in Choreo Connect");
+
+        WsClient wsClient = new WsClient(eventCountThrottleEndpoint, headers);
+        Assert.assertTrue(wsClient.isThrottledWebSocket(throttleCount), "Request not throttled by event count");
+        // TODO: (suksw) Add testcase for client sent event count throttling
```

Comment:
Let's create an issue or link to an already existing issue?

Reply:
Created an issue and linked this PR. https://github.com/wso2/product-microgateway/issues/2767